File: RiotGames/Dto/Match/TeamDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Match;

public sealed record TeamDto
{
    [JsonProperty("bans")]
    public IReadOnlyCollection<BanDto> Bans { get; init; }

    [JsonProperty("objectives")]
    public ObjectivesDto Objectives { get; init; }

    [JsonProperty("teamId")]
    public int TeamId { get; init; }

    [JsonProperty("win")]
    public bool Win { get; init; }
}

File: src/potg/riot/lol/api/match/dto/team.dto.ts
import { Field, ObjectType } from "@nestjs/graphql";
import { BanDto } from "./ban.dto";
import { ObjectivesDto } from "./objectives.dto";

@ObjectType()
export class TeamDto {
  @Field(() => [BanDto])
  bans: BanDto[];
  @Field()
  objectives: ObjectivesDto;
  @Field()
  teamId: number;
  @Field()
  win: boolean;
}
